### PR TITLE
Fix issue #299

### PR DESCRIPTION
**Pull request by ShubhamKanitkar32**
# Description

Fixes issue - https://github.com/roboflow/supervision/issues/299

## Type of change

Please delete options that are not relevant.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

Test script provided under the issue

### CODE REVIEW

File: supervision/detection/utils.py [hunk 1]
```diff
@@ -20,7 +20,10 @@ def polygon_to_mask(polygon: np.ndarray, resolution_wh: Tuple[int, int]) -> np.n
     """
     width, height = resolution_wh
     mask = np.zeros((height, width))
-    cv2.fillPoly(mask, [polygon], color=1)
+
+    # Empty numpy array check
```
Comment (artyaltanzaya) [inline supervision/detection/utils.py:24]: Thank you for addressing the bug; I suggest using if polygon.size > 0: to directly check for an empty array, ensuring against cases with all-zero coordinates.

**Comment by SkalskiP** [comment]
Hi, @ShubhamKanitkar32 👋🏻! Thanks a lot for submitting a PR so fast. We definitely can't allow `supervision` to crash when processing `roboflow` results. The question that I have is: are we fixing the right thing? 

It seems reasonable for me for the `polygon_to_mask` to crash with invalid input. After all... you need at least three distinct points to draw a polygon on a 2d plane. 

Maybe we should move that fix to `sv.Detections.from_roboflow` and check if returned detection is valid? It is a bit surprising that the segmentation result can be an empty list of points and be valid.

**Comment by ShubhamKanitkar32** [comment]
@SkalskiP no problem, happy to contribute!

I agree with you. It makes more sense for it to live under `sv.Detections.from_roboflow()` -> `process_roboflow_result()`

**Comment by SkalskiP** [comment]
Awesome @ShubhamKanitkar32! 🔥 In that case, I have to ask you for two things:
- Accept our CLA. You can read more about it here: https://github.com/roboflow/supervision/pull/300#issuecomment-1684393454. This way, we can merge your PR later and credit you for your contribution.
- Move the conversation to issue: https://github.com/roboflow/supervision/issues/299 and provide me with the example of JSON you pass as an argument to `sv.Detections.from_roboflow()` that brakes the API. I'm interested not only in the list of points (I understand that it is empty) but the whole JSON object you pass.

Once again thanks for helping us to get to the bottom of this problem 🙏🏻

**Comment by SkalskiP** [comment]
Hi @ShubhamKanitkar32 👋🏻! I think we are ready to implement. Take a look at my last comment under your issue: https://github.com/roboflow/supervision/issues/299#issuecomment-1686908979

**Review by SkalskiP — APPROVED** [review]

**Comment by SkalskiP** [comment]
Hi, @ShubhamKanitkar32 I've made some changes to your implementation and added unit tests to make sure it works as expected. I'm merging changes into `develop`. Please check if everything works as expected.

**Comment by SkalskiP** [comment]
btw, @onuralpszr I love per-commit

**Pull request merged**